python-wrapper/tests/test_render.py

```diff
@@ -121,3 +93,20 @@ def test_render_warnings() -> None:
         "relationships. If you need these features, use the canvas renderer by setting the `renderer` parameter",
     ):
         VG.render(max_allowed_nodes=20_000, renderer=Renderer.WEB_GL)
+
+
+def test_render_non_json_serializable() -> None:
+    import datetime
+
+    nodes = [
+        Node(
+            id=0,
+            properties={
+                "non-json-serializable": datetime.datetime.now(),
+            },
+        )
+    ]
+
+    VG = VisualizationGraph(nodes=nodes, relationships=[])
+
+    VG.render()
```

Comment:
you could also write a test which just assert over the serialized json instead of the whole render function

Reply:
Agreed, done as well!